HeterogeneousCore/CUDAUtilities/interface/prefixScan.h

```diff
@@ -0,0 +1,169 @@
+#ifndef HeterogeneousCore_CUDAUtilities_interface_prefixScan_h
+#define HeterogeneousCore_CUDAUtilities_interface_prefixScan_h
+
+#include <cstdint>
+
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCompat.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cuda_assert.h"
+
+#ifdef __CUDA_ARCH__
+template <typename T>
+__device__ void __forceinline__ warpPrefixScan(T const* __restrict__ ci, T* __restrict__ co, uint32_t i, uint32_t mask) {
```

Comment:
I'd move only the classes and functions that are not available only in device code, since that is clearly CUDA-specific.

Reply:
These get usable by CPU code because of `cudaCompat.h`, right?

Reply:
Ah, your logic was the other way around, i.e. keep `__device__` only functions in the global namespace. On the other hand it looks to me that the functions in this file get usable in CPU code because of the inclusion of `cudaCompat.h`.

Reply:
these are inside `#ifdef __CUDA_ARCH__`, so they are ever only compiled for device code

Reply:
Right, thanks for the clarification.